modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+                .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+        Mockito.when(userManager.listUsersWithPost(any(SearchRequest.class),
+                anyMap())).thenThrow(new BadRequestException());
+        SCIMResponse scimResponse = userResourceManager.listWithPOST(RESOURCE_STRING, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_BAD_REQUEST);
+    }
+
+    @DataProvider(name = "dataForUpdateWithPATCH")
+    public Object[][] dataToUpdateWithPATCH() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH, schema, new User());
+        String id = userOld.getId();
+        User userNew = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, schema, new User());
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, "userName", null, userNew, userOld}
+        };
+    }
+
+    @Test(dataProvider = "dataForUpdateWithPATCH")
+    public void testUpdateWithPATCH(String existingId, String scimObjectString,
+                                    String attributes, String excludeAttributes, Object scimNewUserObject,
+                                    Object scimOldUserObject)
+            throws BadRequestException, CharonException, NotImplementedException, NotFoundException {
+
+        User userNew = (User) scimNewUserObject;
+        User userOld = (User) scimOldUserObject;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+
+        abstractResourceManager.when(() -> userManager.getUser(existingId,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(userOld);
+        User validatedUser = (User) ServerSideValidator.validateUpdatedSCIMObject(userOld, userNew, schema);
+        Mockito.when(userManager.updateUser(any(User.class), anyMap(), anyList())).thenReturn(validatedUser);
+        SCIMResponse scimResponse = userResourceManager.updateWithPATCH(existingId, scimObjectString,
+                userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(scimResponse.getResponseMessage());
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_OK);
+        String returnedURI = scimResponse.getHeaderParamMap().get(SCIMConstants.LOCATION_HEADER);
+        String expectedURI = SCIM2_USER_ENDPOINT + "/" + obj.getString("id");
+        Assert.assertEquals(returnedURI, expectedURI);
```

Comment:
better to assert if the patched value is there in the response

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11

Reply:
is this fixed?

Reply:
Return SCIM object been generated as a mock, based on the operation ex - ``add``, ``remove``,  Mock object been generated with relevant modification(s), since it oblivious and no verifying any real procedure, its not been tested as an assertion.